### DISCUSSION TITLE
auto sort gleam.toml depdendencies on `gleam add`

### DISCUSSION THREAD

**Discussion by inoas** [gleam-lang/gleam#3717] [Ideas & suggestions]

**Comment by lpil** [comment]
What would be the motivation here?

**Reply by abdulhakim-alshanqiti** [reply]
probably to for clarity if you have packages that are under simmilar names like


mark_tools_ui
mark_tools_charts


they will be next to each other its probably not needed because most projects have not that many dependancies but might be usefull if Gleam turned out to have too many packages like JavaScript projects have

**Reply by inoas** [reply]
makes it quick to scan the deps lists if you want to manage semver
this is obviously low prio but also probably not tooo hard to do in case people are looking for easy features to impl

we could also have groups separated by full line comments or double new lines?

**Reply by giacomocavalieri** [reply]
I'm not in favour of groups, but I do agree that having the deps in a fixed alphabetical order sounds nice to quickly scan through the deps of a project

**Reply by abdulhakim-alshanqiti** [reply]
yes groups would be strange i haven't seen them anywhere but because Gleam formatter order dependancies in alphabitical order it would make sense that the toml file would do the same thing

**Reply by lpil** [reply]
If we do this we'd need to maintain both comments and groups. It could be challenging to implement.

Another thing to consider is that it would result in very noisy git diffs, which could obscure what's changed in PRs

**Reply by abdulhakim-alshanqiti** [reply]
> yes groups would be strange i haven't seen them anywhere but because Gleam formatter order dependancies in alphabitical order it would make sense that the toml file would do the same thing by not having groups and ordering things alphabiticaly

**Reply by lpil** [reply]
The formatter does support groups for imports. We wouldn't do otherwise here.

**Reply by inoas** [reply]
It does make sense to use comments so these would need to stick around if placed after or ontop of a dep IMHO

**Reply by inoas** [reply]
> Another thing to consider is that it would result in very noisy git diffs, which could obscure what's changed in PRs

- The noise would only happen once and/or if people manually added deps and then used gleam add. If the behaviour is consistent it wouldn't be there.
- There are semantic diff tools who would not report this, such as difftastic I suppose
